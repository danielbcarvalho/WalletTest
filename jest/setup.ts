import 'react-native-gesture-handler/jestSetup';
import { mockDefaultTheme } from '../src/theme/defaultTheme';

jest.mock('styled-components', () => {
  const original = jest.requireActual('styled-components');

  return {
    ...original,
    useTheme: () => {
      return mockDefaultTheme;
    },
  };
});
