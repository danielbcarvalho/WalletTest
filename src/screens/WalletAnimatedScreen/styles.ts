import { SafeAreaView } from 'react-native-safe-area-context';

import styled, { css } from 'styled-components/native';

interface Props {
  top?: boolean;
}

export const Container = styled(SafeAreaView)<Props>`
  flex: 1;
  background-color: ${({ theme }) => theme.pallete.background.default};
  padding: 10px 30px;
  justify-content: center;
  align-items: center;
`;

export const BackgroundAnimated = styled.View<Props>`
  position: absolute;
  height: 230px;
  width: 370px;
  background-color: ${({ theme }) => theme.pallete.background.overlay};
  opacity: 0.2;
  border-bottom-left-radius: 50px;
  border-bottom-right-radius: 50px;

  ${({ top }) =>
    top
      ? css`
          transform: rotate(-39.93deg);
          top: -70px;
          right: 120px;
        `
      : css`
          transform: rotate(139deg);
          bottom: -70px;
          left: 120px;
        `}
`;
