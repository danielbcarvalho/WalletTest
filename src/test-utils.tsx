// TODO: Verificar essa melhoria nos testes
import React from 'react';
import {
  render,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { mockDefaultTheme } from './theme/defaultTheme';

interface IAllThemeProviders {
  children: React.ReactElement;
}
const AllTheProviders = ({ children }: IAllThemeProviders) => {
  return <ThemeProvider theme={mockDefaultTheme}>{children}</ThemeProvider>;
};

function customRender(
  ui: React.ReactElement,
  options = {} as RenderOptions,
): RenderAPI {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

export function rendererCreate(
  component: React.ReactElement,
): ReactTestRenderer {
  return renderer.create(<AllTheProviders>{component}</AllTheProviders>);
}

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
