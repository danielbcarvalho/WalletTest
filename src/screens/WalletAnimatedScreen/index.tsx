import React, { useEffect } from 'react';
import { StyleSheet, Image } from 'react-native';
import Animated, {
  Easing,
  withTiming,
  useSharedValue,
  useAnimatedStyle,
} from 'react-native-reanimated';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';

import { Container } from './styles';
import { ParamsList } from '../../Routes';
import { useNavigation } from '@react-navigation/native';
import { useCardList } from '../../components/hooks/useCardList';

export type WalletAnimatedScreenProps = NativeStackNavigationProp<
  ParamsList,
  'CardList'
>;
const config = {
  duration: 1000,
  easing: Easing.bezier(0.2, 0.2, 0.2, 0.2),
};

function WalletAnimatedScreen() {
  const { isLoading, isError } = useCardList();
  const { navigate } = useNavigation<WalletAnimatedScreenProps>();

  const width = useSharedValue(400);
  const height = useSharedValue(235);

  const animatedSize = useAnimatedStyle(() => {
    return {
      width: withTiming(width.value, config),
      height: withTiming(height.value, config),
      position: 'absolute',
    };
  });

  useEffect(() => {
    const interval = setInterval(() => {
      width.value = width.value === 460 ? 400 : 460;
      height.value = height.value === 500 ? 235 : 500;
    }, 1500);

    return () => {
      clearInterval(interval);
    };
  }, [height, width]);

  useEffect(() => {
    if (!isLoading && !isError) {
      navigate('CardList');
    }
  }, [isLoading, isError, navigate]);

  return (
    <Container>
      <Animated.View
        style={[styles.background, styles.backgroundRotateTop, animatedSize]}
      />
      <Image
        resizeMode="contain"
        style={styles.image}
        source={require('../../../assets/images/animationwallet.png')}
      />
      <Animated.View
        style={[styles.background, styles.backgroundRotateBottom, animatedSize]}
      />
    </Container>
  );
}

const styles = StyleSheet.create({
  background: {
    flex: 1,
    backgroundColor: '#EEEEEE',
    opacity: 0.2,
    borderBottomLeftRadius: 50,
    borderBottomRightRadius: 50,
  },
  backgroundRotateTop: {
    transform: [
      { rotate: '148deg' },
      { translateY: -380 },
      { translateX: 140 },
    ],
  },
  backgroundRotateBottom: {
    transform: [
      { rotate: '-30deg' },
      { translateY: -380 },
      { translateX: 140 },
    ],
  },
  image: {
    width: 115,
    height: 115,
  },
});

export default WalletAnimatedScreen;
