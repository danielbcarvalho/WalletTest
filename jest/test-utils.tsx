import React, { ReactElement, FC } from 'react';
import { render } from '@testing-library/react-native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ThemeProvider } from '../src/theme';

type Options = Parameters<typeof render>[1];

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      cacheTime: Infinity,
    },
  },
});
const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => (
  <ThemeProvider>
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
