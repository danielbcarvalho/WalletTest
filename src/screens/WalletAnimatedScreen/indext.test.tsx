import React from 'react';
import { useNavigation } from '@react-navigation/native';

import { createQueryClientWrapper, render } from '../../../jest/test-utils';

import WalletAnimatedScreen from './';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));
describe('WalletAnimatedScreen', () => {
  let navigateMock: jest.Mock;

  beforeEach(() => {
    navigateMock = jest.fn();
    useNavigation.mockReturnValue({ navigate: navigateMock });
  });

  test('should render correctly', async () => {
    const { toJSON } = render(<WalletAnimatedScreen />, {
      wrapper: createQueryClientWrapper(),
    });

    expect(toJSON()).toMatchSnapshot();
  });
});
