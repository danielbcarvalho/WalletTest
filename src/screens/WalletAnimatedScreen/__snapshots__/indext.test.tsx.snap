// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletAnimatedScreen should render correctly 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#142995",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingBottom": 10,
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 10,
    }
  }
>
  <View
    animatedStyle={
      {
        "value": {
          "height": 235,
          "position": "absolute",
          "width": 400,
        },
      }
    }
    collapsable={false}
    style={
      {
        "backgroundColor": "#EEEEEE",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "flex": 1,
        "height": 235,
        "opacity": 0.2,
        "position": "absolute",
        "transform": [
          {
            "rotate": "148deg",
          },
          {
            "translateY": -380,
          },
          {
            "translateX": 140,
          },
        ],
        "width": 400,
      }
    }
  />
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../assets/images/animationwallet.png",
      }
    }
    style={
      {
        "height": 115,
        "width": 115,
      }
    }
  />
  <View
    animatedStyle={
      {
        "value": {
          "height": 235,
          "position": "absolute",
          "width": 400,
        },
      }
    }
    collapsable={false}
    style={
      {
        "backgroundColor": "#EEEEEE",
        "borderBottomLeftRadius": 50,
        "borderBottomRightRadius": 50,
        "flex": 1,
        "height": 235,
        "opacity": 0.2,
        "position": "absolute",
        "transform": [
          {
            "rotate": "-30deg",
          },
          {
            "translateY": -380,
          },
          {
            "translateX": 140,
          },
        ],
        "width": 400,
      }
    }
  />
</RNCSafeAreaView>
`;
